test/test_basic_logic.py

```diff
@@ -605,7 +605,7 @@ def test_reject_trailers_not_ending_stream(self, frame_factory):
         c = h2.connection.H2Connection()
         c.initiate_connection()
         c.send_headers(1, self.example_request_headers)
-        f = frame_factory.build_headers_frame(self.example_request_headers)
+        f = frame_factory.build_headers_frame(self.example_response_headers)
```

Comment:
Heh, nice catch! That's a good validation that the new code is working as expected!